### PR TITLE
Owls 99916 - Add anti-affinity content by default for the clustered and the non-clustered server pods

### PR DESCRIPTION
**Pull request by ankedia**
Owls 99916 - Changes to add anti-affinity content by default for the clustered and the non-clustered server pods. The PR also removes the affinity contents from all the samples as it's now the default. It updates the "Node Heating problem" FAQ to say that affinity content is added by default beginning with 4.0. The schema description is updated to point to the FAQ document.

Integration test run at - https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/11052
and https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/11056/.

### CODE REVIEW

**Comment by ankedia** [comment]
Converting it to draft PR, I missed adding unit tests for the upgrade path.

File: documentation/staging/content/faq/node-heating.md [hunk 1]
```diff
@@ -45,11 +70,11 @@ serverPod:
...
           labelSelector:
             matchExpressions:
-            - key: "weblogic.domainUID"
+            - key: "security"
               operator: In
               values:
-              - $(DOMAIN_UID)
-          topologyKey: "kubernetes.io/hostname"
+              - "S1"
+          topologyKey: "failure-domain.beta.kubernetes.io/zone"
```
[the remark this answers is not in the document]
Comment (rjeberhard) [reply documentation/staging/content/faq/node-heating.md:77]: I've approved, so this is my only comment. It seems odd to show an example with a "beta" key when we could use "topology.kubernetes.io/zone" or "kubernetes.io/hostname".

**Comment by rjeberhard** [comment]
@rosemarymarano, would you mind please reviewing the documentation changes? Thanks!

**Comment by rosemarymarano** [comment]
@rjeberhard I'm on it.

File: documentation/staging/content/faq/node-heating.md [hunk 2]
```diff
@@ -12,7 +12,7 @@ One solution is to ensure that all necessary container images are available on w
 
 The operator team recommends a different solution that is based on [inter-pod affinity and anti-affinity](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#inter-pod-affinity-and-anti-affinity). This solution has the advantage of both resolving the Node heating problem and of explicitly directing the Kubernetes Scheduler to spread the Pods for WebLogic Server instances from a given cluster or domain more widely across the available Nodes. Inter-pod affinity and anti-affinity are features of the Kubernetes Scheduler that allow the scheduler to choose a Node for a new Pod based on details of the Pods that are already running. For WebLogic Server use cases, the intent will often be for anti-affinity with the Pods for other WebLogic Server instances so that server instances spread over the available Nodes.
 
-To use these features, edit the Domain Custom Resource to add content to the `serverPod` element, in this case at the scope of a cluster, as shown in the following example:
+Beginning with operator version 4.0, the operator adds the following content to the `serverPod` element at the cluster scope in the Domain Custom Resource by default.
```
Comment (rosemarymarano) [inline documentation/staging/content/faq/node-heating.md:15]: Beginning with operator version 4.0, the operator adds the following content to the `serverPod` element at the cluster scope in the Domain Custom Resource by default. -> Beginning with operator version 4.0, by default, the operator adds the following content to the `serverPod` element at the cluster scope in the Domain Custom Resource.

File: documentation/staging/content/faq/node-heating.md [hunk 3]
```diff
@@ -29,12 +29,37 @@ clusters:
...
+Because the `serverPod` element here is scoped to a cluster, the content of the `affinity` element will be added to the Pod generated for each WebLogic Server instance that is a member of this WebLogic cluster. This inter-pod anti-affinity statement expresses a preference that the scheduler select a Node for the new Pod avoiding, as much as possible, Nodes that already have Pods with the label "weblogic.clusterName" and the name of this cluster and with the label "weblogic.domainUID" and the UID of this domain. Note that the `weight` is set to `100`, which is the maximum weight, so that this term will outweigh any possible preference for a Node based on availability of container images.
```
Comment (rosemarymarano) [inline documentation/staging/content/faq/node-heating.md:39]: "weblogic.clusterName" -> `weblogic.clusterName` (Instances where you have double quotes, should be code font, no quotes. I know it was there previously but let's fix it now)
"weblogic.clusterName" and the name of this cluster -> `weblogic.clusterName` and the name of this cluster, (comma)
"weblogic.domainUID" -> `weblogic.domainUID`

File: documentation/staging/content/faq/node-heating.md [hunk 4]
```diff
@@ -29,12 +29,37 @@ clusters:
...
+                matchExpressions:
+                  - key: "weblogic.domainUID"
+                    operator: In
+                    values:
+                    - $(DOMAIN_UID)
+              topologyKey: "kubernetes.io/hostname"
+```
 
-It is possible to express many other scheduling preferences or constraints. The following example similarly expresses an anti-affinity, but changes the test to have all WebLogic Server instances in the domain prefer to run on Nodes where there is not already a Pod for a running instance:
+It is possible to customize the default affinity preferences, and express many other scheduling preferences or constraints. The following example similarly expresses an anti-affinity, but changes the test to have all WebLogic Server instances in the domain prefer to run on Nodes where there is not already a Pod with "security" label with value of "S1".
```
Comment (rosemarymarano) [inline documentation/staging/content/faq/node-heating.md:62]: preferences, and -> preferences and (no comma needed here)
with "security" label with value of "S1". -> with a `security` label with the value of `S1`.

**Review by rosemarymarano — COMMENTED** [review]
Few edits.

File: documentation/staging/content/faq/node-heating.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (ankedia) [reply documentation/staging/content/faq/node-heating.md]: I have changed the topologyKey to "kubernetes.io/hostname". I had taken this policy from one of the unit tests and that had previous topologyKey.

File: documentation/staging/content/faq/node-heating.md (hunk 4 above)
Comment (ankedia) [reply documentation/staging/content/faq/node-heating.md]: Fixed.

File: documentation/staging/content/faq/node-heating.md (hunk 3 above)
Comment (ankedia) [reply documentation/staging/content/faq/node-heating.md]: Done.

**Pull request merged**